src/lib.rs

```diff
@@ -51,12 +53,12 @@ pub use error::Error;
 ///
 /// This is the core type of this crate, and is used to create a new
 /// server and listen for connections.
-pub struct Server {
-    handler: fn(Request<&[u8]>, ResponseBuilder) -> Result<Response<&[u8]>, Error>,
+pub struct Server<T> {
```

Comment:
i prefer to repeat the bounds on the struct as well; as this means you could construct a `Server` that doesn't have the `listen` method!

Reply:
Doing so means `Server` will have to have a `PhantomData` field. Is it worth it?

Also is it fine to have a `marker: PhantomData<&'a ()>` here, or does it have to be `marker: PhantomData<&'a T>`? Both compile just fine, but the latter requires adding the `Sync` trait to `T`.

Reply:
hm, does it? That strikes me as odd....

Reply:
okay, i guess that we shouldn't do this, then

would you mind adding an example in the examples directory that returns owned stuff? After that, this is good to go IMO